### PR TITLE
[test] Update OSU benchmark to include prgenv nvidia

### PR DESCRIPTION
**Pull request by ekouts**
Add `PrgEnv-nvidia` in OSU benchmark and update syntax of the tests

### CODE REVIEW

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py [hunk 1]
```diff
@@ -3,100 +3,139 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
+from reframe.core.decorators import run_after
```
Comment (jjotero) [inline cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:6]: We don't need this, right?

**Review by jjotero — CHANGES_REQUESTED** [review]

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py (hunk 1 above)
Comment (ekouts) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:6]: Good point, I removed it

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py [hunk 2]
```diff
@@ -7,96 +7,134 @@
...
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    descr = 'Alltoall OSU microbenchmark'
+    build_system = 'Make'
+    executable = './osu_alltoall'
+    # The -m option sets the maximum message size
+    # The -x option sets the number of warm-up iterations
+    # The -i option sets the number of iterations
+    executable_opts = ['-m', '8', '-x', '1000', '-i', '20000']
+    valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                           'PrgEnv-intel', 'PrgEnv-nvidia']
```
Comment (jgphpc) [inline cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:23]: For some reason, reframe does not unload PrgEnv-cray in my scripts, only:
```
module load PrgEnv-nvidia
module load cdt/21.05
```
Do you get the same ?

Comment (ekouts) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:23]: Strange, I am looking now at the stage directory of one of the tests (in  dom) and it looks like it does
```
...
module load daint-gpu
module unload PrgEnv-cray
module load PrgEnv-nvidia
module load cdt/21.05
...
```
Where did you run it?

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py [hunk 3]
```diff
@@ -194,36 +235,37 @@ def __init__(self):
...
-                'latency': (2.33, None, 0.15, 'us')
-            },
-            'pilatus:mc': {
-                'latency': (2.33, None, 0.15, 'us')
-            },
-            # keeping as reference:
-            # 'monch:compute': {
-            #     'latency': (1.27, None, 0.1, 'us')
-            # },
+    executable = './p2p_osu_latency'
```
Comment (jgphpc) [inline cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:238]: valid_systems missing ?

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py [hunk 4]
```diff
@@ -233,70 +275,102 @@ def __init__(self):
...
+        self.build_system.cppflags = ['-D_ENABLE_CUDA_']
+
+    @run_before('compile')
+    def set_modules(self):
         if self.current_system.name in ['daint', 'dom']:
             self.num_gpus_per_node  = 1
-            self.modules = ['craype-accel-nvidia60']
             self.variables = {'MPICH_RDMA_ENABLED_CUDA': '1'}
+            if self.current_environ.name == 'PrgEnv-nvidia':
+                self.modules = ['cdt-cuda/21.05']
```
Comment (jgphpc) [inline cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:313]: I did not know before (sorry) but can you try with adding `cudatoolkit/21.3_11.2` to the list of modules and remove the ldflags to see if it works ?

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py [hunk 5]
```diff
@@ -233,70 +275,102 @@ def __init__(self):
...
```
Comment (jgphpc) [inline cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:363]: same comment

**Review by jgphpc — APPROVED** [review]
lgtm

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py (hunk 3 above)
Comment (ekouts) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:238]: you are right, I deleted by mistake, thanks!

File: cscs-checks/microbenchmarks/mpi/osu/osu_tests.py (hunk 4 above)
Comment (ekouts) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:313]: It is working with `cudatoolkit/21.3_11.2`, I will update the test 👍 The test is also passing without `cdt-cuda/21.05`, should I remove it as well?

Comment (jgphpc) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py:313]: I would say yes, remove it.

Comment (jjotero) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py]: But having the `ldflags` seems a more maintainable solution for the future than hardcoding a specific `cudatoolkit` version, right?

Comment (jgphpc) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py]: I agree (but then who will test the cudatoolkit/cray wrapper ?).
Both work, what do you think the user will prefer ?

Comment (jjotero) [reply cscs-checks/microbenchmarks/mpi/osu/osu_tests.py]: Not sure, but we need to stay consistent here. Now some tests do the `ldflags` by hand and others use the `cudatoolkit` wrapper. If you want to check that all those work, we should probably write a check under `prgenv`, and not test that under with a `microbenchmark`.

**Review by jjotero — APPROVED** [review]

**Comment by jjotero** [comment]
This should get the PEP issues fixed before merging.

**Pull request merged**